lib/reqwestRequest.js

```diff
@@ -10,4 +10,20 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-define(['q', 'reqwest'], function(Q, reqwest) {
\ No newline at end of file
+var reqwest = require('reqwest');
+
+function reqwestRequest(method, url, args) {
+  var r = reqwest({
+    url: url,
+    method: method,
+    headers: args.headers,
+    data: JSON.stringify(args.data),
+    withCredentials: true
+  })
+  .then(function() {
```

Comment:
Does reqwest not return something that we can use to return what we want (i.e. why the var r = foo; return r?):
```javascript
return reqwest({ stuff }).then(function (resp) {
  // does resp not give us whatever is in r.request?
});
```